docs/src/main/paradox/bootstrap/index.md

```diff
@@ -86,6 +86,71 @@ is strictly defined and is as follows:
+that your deployment is robust.
+
+Since the target environments for this module are dynamic, that is, instances can come and go, failure needs to be
+considered. The following configuration will result in your application being shut down after 30 seconds if it is unable to
+join the discovered seed nodes. In this case, the orchestrator (i.e. Kubernetes or Marathon) will restart your node
+and the operation will (presumably) eventually succeed. You'll want to specify the following in your `application.conf` file:
+
+@@snip [CompileOnly.scala](/bootstrap-demo/local/src/main/resources/application.conf) { #coorindated-shutdown }
+
+### Rolling updates
+
+#### Graceful shutdown 
+
+Akka Cluster can handle hard failures using a downing provider such as Lightbend's split brain resolver discussed below.
+However this should not be relied upon for regular rolling redeploys. Features such as `ClusterSinglton`s and `ClusterSharding`
+can safely restart actors on new nodes far quicker when it is certain that a node has shutdown rather than crashed. 
+
+Graceful leaving will happen with the default settings as it is part of Coordinated Shutdown. Just ensure that
+a node is sent a `SIGTERM` and not a `SIGKILL`. Environments such as Kubernetes will do this, it is important to ensure 
+that if JVM is wrapped with a script that it forwards the signal. Upon receiving a `SIGTERM` Coordinated shutdown will:
+
+* Perform a `Cluster(system).leave` on its self
+* The `Leader` will then move the node to `Exiting` and then `Removed` allowing any shards to be shutdown gracefully
+  and `ClusterSingleton`s to be migrated if this was the oldest node
+
+#### Number of nodes to redeploy at once
+
+Akka bootstrap requires a `stable-period` where service discovery returns a stable set of contact points. When doing rolling
+updates it is best to wait for a node (or group of nodes) to finish joining the cluster before adding and removing other nodes.
+
+#### Cluster singletons
+
+`ClusterSingleton`s run on the oldest node in the cluster. To avoid singletons moving during every node deployment it is advised
+to start a rolling redeploy starting at the newest node. Then `ClusterSingleton`s only move once. This is the default behaviour 
+for Kubernetes deployments.
+
+### Split brains and ungraceful shutdown
+
+Nodes can crash causing cluster members to become unreachable. This is a tricky problem as it is not
+possible to distinguish between a network partition and a node failure. To rectify this in an automated manner,
+Lightbend provides [Split Brain Resolver](https://developer.lightbend.com/docs/akka-commercial-addons/current/split-brain-resolver.html)
+as a feature of the Lightbend Subscription. This module has a number of strategies that can ensure that the cluster
+continues to function during network partitions and node failures.
+
 ## Bootstrap Recipes
 
 To see how to configure and use bootstrap in various environments such as Kubernetes, see @ref[recipes](recipes.md).
```

Comment:
recipes.md is not in any toc and therefore the left side of that page looks strange

Reply:
Fixed